libs/muddle/src/muddle_endpoint_fake.cpp

```diff
@@ -0,0 +1,180 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "muddle_endpoint_fake.hpp"
+
+using fetch::muddle::FakeMuddleEndpoint;
+using fetch::muddle::NetworkId;
+
+using Packet          = FakeMuddleEndpoint::Packet;
+using PacketPtr       = FakeMuddleEndpoint::PacketPtr;
+using Address         = FakeMuddleEndpoint::Address;
+using Response        = FakeMuddleEndpoint::Response;
+using SubscriptionPtr = FakeMuddleEndpoint::SubscriptionPtr;
+using AddressList     = FakeMuddleEndpoint::AddressList;
+using AddressSet      = FakeMuddleEndpoint::AddressSet;
+
+// Convenience function - note shares similarities with real muddle
+PacketPtr FormatPacket(Packet::Address const &from, NetworkId const &network, uint16_t service,
+                       uint16_t channel, uint16_t counter, uint8_t ttl,
+                       Packet::Payload const &payload)
```

Comment:
`from` and `payload` can probably be passed by-value.

Reply:
Yup